src/redux/actions/readPizza.ts

```diff
@@ -0,0 +1,17 @@
+import { PizzaPlace } from '../models';
+import axios from "axios";
+import { JsonDecoder } from "ts.data.json";
+import { pizzaPlacesDecoder } from "../decoders/pizzaDecoder";
+
+async function decodeResponse<T>(decoder: JsonDecoder.Decoder<T[]>, responseData: any): Promise<T[]> {
+  return decoder.decodePromise(responseData);
+}
+
+export async function readPizzaPlaces(): Promise<PizzaPlace[]> {
+  return new Promise((resolve, reject) => {
+    axios
```

Comment:
I think this is a good exercise to go through for this learning project in order to simulate pulling data from a server, but I just wanted to mention that in "real life" if we had something as small as `pizza_places.json` is we'd just include it directly in the bundle and not bother to fetch it separately.

Reply:
Gotcha, this is good to know!